deployment/data-streams/changeset/mcms/deploy_and_transfer_mcms.go

```diff
@@ -0,0 +1,107 @@
+package mcms
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	"github.com/smartcontractkit/chainlink/deployment"
+	commonChangesets "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	commontypes "github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/metadata"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/changeset/types"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/utils"
+	"github.com/smartcontractkit/chainlink/deployment/data-streams/utils/mcmsutil"
+)
+
+var DeployAndTransferMCMSChangeset = deployment.CreateChangeSet(deployAndTransferMcmsLogic, deployAndTransferMcmsPrecondition)
+
+type DeployMCMSConfig struct {
+	ChainsToDeploy []uint64
+	Ownership      types.OwnershipSettings
+	Config         commontypes.MCMSWithTimelockConfigV2
+}
+
+func (cc DeployMCMSConfig) GetOwnershipConfig() types.OwnershipSettings {
+	return cc.Ownership
+}
+
+func deployAndTransferMcmsLogic(e deployment.Environment, cc DeployMCMSConfig) (deployment.ChangesetOutput, error) {
+	cfgByChain := make(map[uint64]commontypes.MCMSWithTimelockConfigV2)
+	for _, chain := range cc.ChainsToDeploy {
+		cfgByChain[chain] = cc.Config
+	}
+
+	mcmsOut, err := commonChangesets.DeployMCMSWithTimelockV2(e, cfgByChain)
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("failed to deploy MCMS: %w", err)
+	}
+
+	// CallProxy has no owner, RBACTimelock has an "admin" setting in place of owner
+	transferContracts := []deployment.ContractType{
+		commontypes.ProposerManyChainMultisig,
+		commontypes.BypasserManyChainMultisig,
+		commontypes.CancellerManyChainMultisig,
+	}
+
+	// DeployMCMSWithTimelockV2 currently does not use the DataStore
```

Comment:
It might be good to drop a ticket and come back to this later, so we don't forget to do it.

Reply:
Hopefully will be handled by DPA / CLD team

Reply:
Dropped request